app/src/main/java/com/eveningoutpost/dexdrip/Home.java

```diff
@@ -3159,13 +3162,35 @@ private void displayCurrentInfoFromReading(BgReading lastBgReading, boolean pred
         }
     }
 
+    // This function is needed in hebrew in order to allow printing the text (for exapmle) -3 mg/dl
+    // It seems that without this hack, printing it is impossibale. Android will print something like:
+    // -mg/dl 3 or mg/dl 3- or  mg/dl -3. (but never -3 mg/dl)
+    // The problem seems to happen becaud of the extra 0x0a char that somehow gets to the buffer.
+    // Since this text has high visabilty, I'm doing it. Will not be done in other places.
+    private void HebrewAppendDisplayData() {
+        // Do the append for the hebrew language
+         String original_text = notificationText.getText().toString();
+        Log.d(TAG, "original_text = " + HexDump.dumpHexString(original_text.getBytes()));
+        if(original_text.length() >=1 && original_text.charAt(0) == 0x0a) {
+            Log.d(TAG,"removing first and appending " + display_delta);
+            notificationText.setText(display_delta + "  " + original_text.substring(1));
+        } else {
+            notificationText.setText(display_delta + "  " + original_text);
+        }
+    }
+
```

Comment:
This glitch is a bit worrying. Do we have any more information about it?

Reply:
Not too much information, this is my observation of what is happening there.
I can try and remove this from this PR and send it in another pr, that will not have hebrew specific code.
Please also note that this is happening for the big tablets, so in any case only very few devices will be affected by this.